### PR TITLE
Fix usage of prepublishOnly

### PR DESCRIPTION
**Pull request by Shinigami92**
## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

The `prepublishOnly` script was misunderstood and not correctly used

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "coverage:html": "nyc report --reporter=html",
     "format": "prettier --loglevel silent --write .; prettierx --write .",
     "lint": "eslint .",
-    "lint:fix": "eslint . --fix"
+    "lint:fix": "eslint . --fix",
+    "prepublishOnly": "npm run test:prod"
```
Comment (jerome-benoit) [inline package.json:21]: It should be `npm run build:prod`. Tests are already run in the dependent job.

Comment (Shinigami92) [reply package.json:21]: Okay

Comment (Shinigami92) [reply package.json]: Personally I find it a bit dangerous to not test before publishing and just rely on the GitHub Actions workflow script, but maybe we will never create a release manually

File: .github/workflows/npmpublish.yml [hunk 2]
```diff
@@ -35,16 +35,10 @@ jobs:
...
           registry-url: https://registry.npmjs.org/
 
-      - name: Install
-        run: npm ci
-
-      - name: Build
-        run: npm run build:prod
-
       - name: Install Production
         run: npm ci --production
```
Comment (jerome-benoit) [inline .github/workflows/npmpublish.yml:39]: I think that will no work, you need the dev dependencies to build. That's why first `npm ci` and then build and then `npm ci --production`

**Review by jerome-benoit — APPROVED** [review]

**Review by jerome-benoit — APPROVED** [review]

**Pull request merged**